### DISCUSSION TITLE
Move spies to dedicated Feature?

### DISCUSSION THREAD

**Discussion by HannesWell** [eclipse-pde/eclipse.pde#415] [Ideas]
As part of their migration to PDE the E4 spies have been added to the `org.eclipse.pde` feature:
https://github.com/eclipse-pde/eclipse.pde/blob/fc645e8682da8e853f21451e9833cabd9918337e/features/org.eclipse.pde-feature/feature.xml#L162-L216

Consequently everybody that uses the `org.eclipse.pde` feature now gets them installed if it is wanted or not.
Since the spies are not a core functionality of PDE I think it would be good to move them to a dedicated feature (for example `org.eclipse.pde.spies`) so that one can choose to install them in a product or not.

@vogella and @opcoach, you were the main drivers behind the migration, what do you think?

**Comment by opcoach** [comment]
Hi, 

I didn't react here about this demand, but I remember to have a talk about it in another discussion. 

Spies are necessary when we develop either E3 or E4 applications. There are necessary to explore the content of the E4 engine and must not be separated in term of delivery.  

I totally disagree with the fact you mentioned : "Since the spies are not a core functionality of PDE " ...  

When you are developing a RCP application, you create plugins, and you should create E4 plugins... but even if you don't, your plugins will run on the E4 engine... that manages application model,  injection context, E4 events... and the spies have been done to show all of these concepts in their own  dashboards.

I installed the 2023-09 Eclipse for RCP and RAP developers, with the PDE, and I unfortunately see that the spies are not provided... while they were in there 2023-06... Where are they now ? 

People use to use the market place... but it has not been updated, and it  goes to MY website (opcoach.com), where I provided a temporary patch between the 2021-12 and 2022-03 releases because we were late to migrate all the spies for the 2022-03... some of them were in the PDE and others where isolated on my p2 web site (opcoach.com). That was a temporary patch to install the missing spies. 

Now they are still available on my p2 repository and proposed for the 2023-09... Not good...  I will change this, right now,  to limit the download only up to  2021-12 and not for newer releases... 

SO I found them using 'Install new software' and use the 2023-09 and look for spies..  by installing a 'regular' feature... 

I didn't find any issue that mentioned this has been removed from the official release ... except the issue #472  that just say it has been move to a dedicated feature...  For me, this feature is ok, but it must still be delivered with the PDE ... ! 

Sorry for this late comment, but it was so straight forward for RCP developers that I couldn't imagine to face this situation... !

**Comment by merks** [comment]
As you noticed, its on the train:

![image](https://github.com/eclipse-pde/eclipse.pde/assets/208716/c1073909-a872-466c-9bb8-6fb46526961f)

The Eclipse SDK feature includes it.  So it's not removed at all.  I think it's just accidentally not included in places where it probably should be included such as the RCP package and probably also in the Committers package.  For that purpose, it's best to open an issue here:

https://github.com/eclipse-packaging/packages

The best the Platform can/should do is make it available to the packages, which is does:

![image](https://github.com/eclipse-pde/eclipse.pde/assets/208716/c776c9f7-d7b0-464d-94c8-838b292e41ac)

**Comment by opcoach** [comment]
Ok I will open it . 

The problem is that the market place does not filter properly the version... :  

Even if the 0.19 of spies is only defined for the 2021-12, it is still proposed for the 2023-09 !! Don't understand why : 

![image](https://github.com/eclipse-pde/eclipse.pde/assets/1716878/0228bb16-b02f-407a-8263-0493e85a7653)

**Reply by merks** [reply]
Looking at the health report

https://www.eclipse.org/setups/marketplace/?id=3116105

Looking at the raw metadata for the definition, I'm not sure why you would expect something different from what you see:

https://marketplace.eclipse.org/content/e4-spies/api/p

**Comment by opcoach** [comment]
Spies are in the Eclipse for committer. They are missing only in RCP and RAP developers.

**Reply by jonahgraham** [reply]
Thanks @opcoach for filing the issue in packages, xref for anyone else who wants to follow along https://github.com/eclipse-packaging/packages/issues/68
